detox/src/android/core/NativeMatcher.js

```diff
@@ -1,27 +1,42 @@
+const { DetoxRuntimeError } = require('../../errors');
 const invoke = require('../../invoke');
 const DetoxMatcherApi = require('../espressoapi/DetoxMatcher');
 
 class NativeMatcher {
+  static _assertMatcher(matcher) {
+    if (!(matcher instanceof NativeMatcher)) {
+      throw new DetoxRuntimeError(`Expected a matcher, got ${typeof matcher}`);
```

Comment:
Nah, use something like `require('util').inspect()`. `typeof object` is not interesting

Reply:
https://github.com/wix/Detox/pull/3980/files#diff-af99071757b1d50b8db1299cb540dd373a3d6346b826447a408b34007fb0115aR10